main.py

```diff
@@ -60,51 +70,84 @@ def scrape_table(headers):
     browser_lib.wait_for_condition(condition=condition, timeout=timedelta(seconds=20))
 
-    table = browser_lib.find_element('//*[@id="investments-table-object"]/tbody')
-    links = [link.get_attribute("href") for link in table.find_elements_by_tag_name("a")]
+    table = browser_lib.find_element('id:investments-table-object >> tag:tbody')
     table_data = [headers]
+    links = []
+    equalities = {}  # dict with "UII": "Name of investment"
 
     for row in table.find_elements_by_tag_name("tr"):
         cols = row.find_elements_by_tag_name("td")
         data_set = []
+
         for data in cols:
             data_set.append(data.text)
+            a_element = data.find_elements_by_tag_name("a")
+            if a_element:
+                links.append(a_element[0].get_attribute("href"))
+
         table_data.append(data_set)
+        equalities[data_set[0]] = data_set[2]
 
-    write_to_excel(sheetname="Individual Investments", bookname="spending.xlsx", content=table_data)
+    write_to_excel(sheetname="Individual Investments", bookname="spending.xlsx", content=table_data, method="append")
 
     for link in links:
-        browser_lib.go_to(link)
-        element = '//*[@id="business-case-pdf"]/a'
-        browser_lib.wait_until_element_is_visible(element, timeout=timedelta(seconds=10))
-        browser_lib.click_element(element)
-        browser_lib.assign_id_to_element(locator=element, id="USERFLAG")
-        condition = "return document.getElementById('USERFLAG').getAttribute('aria-busy') == \"false\""
-        browser_lib.wait_for_condition(condition=condition, timeout=timedelta(seconds=10))
+        file = download_file(link)
+        compare_values(equalities, file)
+
+
+def download_file(link):
+    browser_lib.go_to(link)
+    element = 'id:business-case-pdf >> tag:a'
+
+    browser_lib.wait_until_element_is_visible(element, timeout=timedelta(seconds=20))
+    browser_lib.click_element(element)
+    browser_lib.assign_id_to_element(locator=element, id="USERFLAG")
+
+    condition = "return document.getElementById('USERFLAG').getAttribute('aria-busy') == \"false\""
+    browser_lib.wait_for_condition(condition=condition, timeout=timedelta(seconds=20))
+
+    time.sleep(3)  # wait till file is downloaded
+
+    return get_recent_file("output")
+
+
+def get_recent_file(path):
+    files = os.listdir(path)
+    paths = [os.path.join(path, basename) for basename in files]
+    return max(paths, key=os.path.getctime)
+
+
+def compare_values(equalities, file):
+    text = pdf.get_text_from_pdf(file)[1]
+    investment = re.search(r"1\. Name of this Investment: (.*)2\.", text).group(1)
+    uii = re.search(r"2\. Unique Investment Identifier \(UII\): (.*)Section B", text).group(1)
+
+    if equalities[uii] == investment:
+        logging.info(uii, investment, "EQUAL")
+    else:
+        logging.info(uii, investment, "NOT EQUAL")
```

Comment:
**Logs are not shown:**
![image](https://user-images.githubusercontent.com/36151848/150703555-1d961808-3cfa-41a2-870e-ec91832111eb.png)

Reply:
Depends on environment. Set it like warning instead of info for assuarance in every environment